File: SYP.Web/Modules/Catalog/ProductPacking/ProductPackingColumns.cs
namespace SYP.Catalog.Columns;

[ColumnsScript("Catalog.ProductPacking")]
[BasedOnRow(typeof(ProductPackingRow), CheckNames = true)]
public class ProductPackingColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    [EditLink]
    public string Name { get; set; }
    public int Quantity { get; set; }
    public bool IsActive { get; set; }
    public DateTime InsertDate { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Catalog/ProductPackingColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductPackingRow } from "./ProductPackingRow";

export interface ProductPackingColumns {
    Id: Column<ProductPackingRow>;
    Code: Column<ProductPackingRow>;
    Name: Column<ProductPackingRow>;
    Quantity: Column<ProductPackingRow>;
    IsActive: Column<ProductPackingRow>;
    InsertDate: Column<ProductPackingRow>;
}

export class ProductPackingColumns extends ColumnsBase<ProductPackingRow> {
    static readonly columnsKey = 'Catalog.ProductPacking';
    static readonly Fields = fieldsProxy<ProductPackingColumns>();
}
